File: tests/slicing/switch.ml
(* ledit bin/toplevel.top -deps tests/slicing/switch.c
* *)

include LibSelect ;;

let main _ =
  test_select_data "main" "x";
  test_select_data "main" "y";
  test_select_data "main" "z"
let () = Db.Main.extend main

File: tests/slicing/switch.c
/*  run.config
   EXECNOW: make -s tests/slicing/switch.cmxs
   CMD: @frama-c@ -load-module tests/slicing/libSelect.cmxs -load-module tests/slicing/switch.cmxs
   OPT: -check -deps -journal-disable
*/
int main (char choix) {
  int x = 0, y = 0, z = 0;
  switch (choix) {
    case 'a' : x = 1; break;
    case 'b' : x = 2; y = 1; break;
    case 'c' :
    case 'd' : y = 2; break;
  }
  z++;
  return x;
}
